app/services/onlinetesting/Phase2TestService.scala

```diff
@@ -169,20 +172,27 @@ trait Phase2TestService extends OnlineTestService with Phase2TestConcern with Sc
   }
 
   override def registerAndInviteForTestGroup(applications: List[OnlineTestApplication])
-    (implicit hc: HeaderCarrier, rh: RequestHeader): Future[Unit] =
-    registerAndInviteForTestGroup(applications, getRandomSchedule()).flatMap { candidatesToProgress =>
+                                            (implicit hc: HeaderCarrier, rh: RequestHeader): Future[Unit] = {
+    val (scheduleName, schedule) = getRandomScheduleWithName()
+    registerAndInviteForTestGroup(applications, schedule).flatMap { candidatesToProgress =>
       eventSink {
-        Future.successful(
+        Future.successful {
           candidatesToProgress.map(candidate => {
-            audit("Phase2TestInvitationProcessComplete", candidate.userId)
-            DataStoreEvents.OnlineExerciseResultSent(candidate.applicationId)
-          })
-        )
+            DataStoreEvents.OnlineExerciseResultSent(candidate.applicationId) ::
+              AuditEvents.Phase2TestInvitationProcessComplete(Map(
+                "userId" -> candidate.userId,
+                "absoluteTime" -> s"${calculateAbsoluteTimeWithAdjustments(candidate)}",
+                "scheduleName" -> scheduleName)) ::
+              Nil
+          }).flatten
+        }
       }
     }
+  }
 
   private def registerAndInviteForTestGroup(applications: List[OnlineTestApplication], schedule: Phase2Schedule)
-    (implicit hc: HeaderCarrier, rh: RequestHeader): Future[List[OnlineTestApplication]] = applications match {
+                                           (implicit hc: HeaderCarrier, rh: RequestHeader): Future[List[OnlineTestApplication]]
+  = applications match {
```

Comment:
I feel like the equals is a bit weird on this line, obviously it's valid - best argument i have is that it's inconsistent with the rest of the codebase